### DISCUSSION TITLE
Optimise for target return and target risk

### DISCUSSION THREAD

**Discussion by thatguy002** [dcajasn/Riskfolio-Lib#11] [Q&A]
Hi all

I've just started to use riskfolio-lib after looking at pyportfoliopt and settled on riskfolio-lib.

I am trying to minimise risk for a given target return as well as the expected return for a given level of risk. For example I would like to minimise the portfolio weights for a return of 6% or 10%. Is it also possible to find an expected return for a std dev of 4% or 20%? This seems to be straight forward with pyportfoliopt.

Any help is greatly appreciated. 

Thanks

**Comment by dcajasn** [comment]
Hi, 

Thanks to support my work. Let me tell you that I forgot to add an example using constraints on maximum value for some risk measures. On the other hand, I've just added the option to include constraints on minimum portfolio return. Due to Riskfolio-Lib is based on cvxpy, we need to follow the disciplined convex programming conventions, this means that we only can put bounds on minimum return and maximum risk (return >= level_1 and risk <= level_2). The advantage of Riskfolio-Lib is that we can add bounds on several risk measures at the same time. The new example shows how to find the portfolio that maximizes sharpe ratio with constraints on return, CVaR and Max Drawdown, also shows how these constraints work plotting efficient frontier using Std. Dev., CVaR and Max Drowdown in risk dimension.

Here is the link of the new example https://github.com/dcajasn/Riskfolio-Lib/blob/master/examples/Tutorial%2021.ipynb. If you want to see it in nbviewer, the link is available on documentation.

I would appreciate if you recommend this library to your friends and colleagues.

Best,
Dany

**Reply by thatguy002** [reply]
Hi Dany, thanks for the reply.

I have been over the new example from the link you sent. It seems that when I enter a value for "port.lowerret = 0.16 / 252" and run the efficient frontier graph there is no difference from the efficient frontier graph run without the "port.lowerret" constraint. This leads me to believe that the constraint is being ignored. When I look at the available arguments I can enter for 
"port = pf.Portfolio("Arguments") there is no "lowerret=" option.

Any ideas as to what is going on? I hope my explanation makes sense.

Regards,

**Reply by dcajasn** [reply]
First, you must update the version of Riskfolio-Lib to latest version (0.1.5). I added the property lowerret in this version. Tell me if you still have problems after updating.

**Reply by thatguy002** [reply]
Its working, thank you
